lib/logstash/inputs/syslog.rb

```diff
@@ -74,30 +77,70 @@ class LogStash::Inputs::Syslog < LogStash::Inputs::Base
   #
   config :locale, :validate => :string
 
-  public
-  def register
-    @metric_errors = metric.namespace(:errors)
+  def initialize(*params)
+    super
+
+    @priority_key = ecs_select[disabled:'priority', v1:'[log][syslog][priority]']
+    @facility_key = ecs_select[disabled:'facility', v1:'[log][syslog][facility][code]']
+    @severity_key = ecs_select[disabled:'severity', v1:'[log][syslog][severity][code]']
+
+    @facility_label_key = ecs_select[disabled:'facility_label', v1:'[log][syslog][facility][name]']
+    @severity_label_key = ecs_select[disabled:'severity_label', v1:'[log][syslog][severity][name]']
+
+    @host_key = ecs_select[disabled:'host', v1:'[host][ip]']
+
+    @grok_pattern ||= ecs_select[
+        disabled:"<%{POSINT:#{@priority_key}}>%{SYSLOGLINE}",
+        v1:"<%{POSINT:#{@priority_key}:int}>%{SYSLOGLINE}"
+    ]
 
     @grok_filter = LogStash::Filters::Grok.new(
-      "overwrite" => @syslog_field,
-      "match" => { @syslog_field => @grok_pattern },
-      "tag_on_failure" => ["_grokparsefailure_sysloginput"],
+        "overwrite" => @syslog_field,
+        "match" => { @syslog_field => @grok_pattern },
+        "tag_on_failure" => ["_grokparsefailure_sysloginput"],
+        "ecs_compatibility" => ecs_compatibility # use ecs-compliant patterns
     )
 
+    @grok_filter_exec = ecs_select[
+        disabled: -> (event) { @grok_filter.filter(event) },
+        v1: -> (event) {
+          event.set('[event][original]', event.get(@syslog_field))
+          @grok_filter.filter(event)
+          event.set('[service][type]', 'system') unless event.include?('[service][type]')
```

Comment:
a hard-coded `service.type` default ('aligned' with Beats) for now.
making this configurable requires us to decide how much configuration we would allow e.g.:
```
  syslog {
    service_id => '...'
    service_name => 'ingest-node-1'
    service_type => 'elasticsearch'
  }
```
in general it should be fine to assume non-specific system level logs.

Reply:
The idea to let `service_type` be a configuration option is nice, but do you think we risk an overkill? Do you think it's useful to the user or it's a just something that is used in 1% of the cases? (Unforntunatly I haven't the response :-()

Reply:
have confirmed already that having `service.type` (with a sensible default such as `system`) makes sense here.
my worry was mostly introducing one option but not the others, but I think we could introduce it as un-documented ...
as a compromise whether users would use this ECS only option and would want the option to also set others.